apps/serdes/Deserializer.cpp

```diff
@@ -0,0 +1,608 @@
+#include "Deserializer.h"
+#include <fstream>
+#include <iostream>
+
+std::string Deserializer::deserialize_string(const flatbuffers::String *str) {
+    return str->str();
+}
+
+Halide::Type Deserializer::deserialize_type(const Halide::Serdes::Type *type) {
+    // bits
+    int bits = type->bits();
+
+    // lanes
+    int lanes = type->lanes();
+
+    // code
+    Halide::Serdes::TypeCode code_deserialized = type->code();
+
+    halide_type_code_t code = halide_type_uint;
+    switch (code_deserialized) {
+    case Halide::Serdes::TypeCode::TypeCode_Int:
+        code = halide_type_int;
+        break;
+    case Halide::Serdes::TypeCode::TypeCode_UInt:
+        code = halide_type_uint;
+        break;
+    case Halide::Serdes::TypeCode::TypeCode_Float:
+        code = halide_type_float;
+        break;
+    case Halide::Serdes::TypeCode::TypeCode_Handle:
+        code = halide_type_handle;
+        break;
+    case Halide::Serdes::TypeCode::TypeCode_BFloat:
+        code = halide_type_bfloat;
+        break;
+    }
+
+    return Halide::Type(code, bits, lanes);
+}
+
+Halide::Internal::Function Deserializer::deserialize_function(const Halide::Serdes::Func *function) {
+    // name
+    std::string name = deserialize_string(function->name());
+
+    // origin_name
+    std::string origin_name = deserialize_string(function->origin_name());
+
+    // output_types
+    std::vector<Halide::Type> output_types;
+    auto output_types_serialized = function->output_types();
```

Comment:
Possibly-ignorant question (since I don't know Flatbuffers well): do we need to make a copy here just to do a read-only iteration? ie, can we just do `for (const auto &type : *function->output_types()) {` instead?

Reply:
We can! I believe this line was written when I was still learning how to use the flatbuffers API, you can see I wrote read-only iteration style code in later code when deserializing other structs.